components/brave_wallet/browser/brave_wallet_prefs_unittest.cc

```diff
@@ -0,0 +1,42 @@
+#include "brave/components/brave_wallet/browser/brave_wallet_prefs.h"
+
+#include "base/test/scoped_feature_list.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_prefs.h"
+#include "brave/components/brave_wallet/browser/pref_names.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/testing_pref_service.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(BraveWalletPrefsTest, MigratesCryptoWalletsPrefToBraveWallet) {
+  TestingPrefServiceSimple prefs;
+  prefs.registry()->RegisterIntegerPref(
+      kDefaultEthereumWallet,
+      static_cast<int>(mojom::DefaultWallet::AskDeprecated));  // default unused
+
+  // Changes CryptoWalletsDeprecated to BraveWalletPreferExtension
+  prefs.SetInteger(
+      kDefaultEthereumWallet,
+      static_cast<int>(mojom::DefaultWallet::CryptoWalletsDeprecated));
+  MigrateCryptoWalletsPrefToBraveWallet(&prefs);
+  EXPECT_EQ(prefs.GetInteger(kDefaultEthereumWallet),
+            static_cast<int>(mojom::DefaultWallet::BraveWalletPreferExtension));
+
+  // Doesn't change BraveWallet pref
+  prefs.SetInteger(kDefaultEthereumWallet,
+                   static_cast<int>(mojom::DefaultWallet::BraveWallet));
+  MigrateCryptoWalletsPrefToBraveWallet(&prefs);
+  EXPECT_EQ(prefs.GetInteger(kDefaultEthereumWallet),
+            static_cast<int>(mojom::DefaultWallet::BraveWallet));
+
+  // Doesn't change None pref
+  prefs.SetInteger(kDefaultEthereumWallet,
+                   static_cast<int>(mojom::DefaultWallet::None));
+  MigrateCryptoWalletsPrefToBraveWallet(&prefs);
+  EXPECT_EQ(prefs.GetInteger(kDefaultEthereumWallet),
+            static_cast<int>(mojom::DefaultWallet::None));
```

Comment:
Miss a `BraveWalletPreferExtension` case for the non change test and we can deduplicate these three cases in a loop.